cmd/check_spec.go

```diff
@@ -122,58 +92,448 @@ func newCheckSpec(ctx context.Context) (*checkSpec, error) {
+	_, _, _, err = cs.onlineFetcher.AccountBalanceRetry(ctx, emptyNetwork, emptyAcct, emptyPartBlock, emptyCur)
+	validateErrorObject(err, output)
+
+	if isUTXO() {
+		printInfo("%v\n", "sending request to /account/coins ...")
+		_, _, _, err = cs.onlineFetcher.AccountCoinsRetry(ctx, emptyNetwork, emptyAcct, false, emptyCur)
+		validateErrorObject(err, output)
+	} else {
+		printInfo("%v\n", "skip /account/coins for account based chain")
+	}
+
+	printInfo("%v\n", "sending request to /block ...")
+	_, err = cs.onlineFetcher.BlockRetry(ctx, emptyNetwork, emptyPartBlock)
+	validateErrorObject(err, output)
+
+	printInfo("%v\n", "sending request to /block/transaction ...")
+	emptyTx := []*types.TransactionIdentifier{}
+	emptyBlock := &types.BlockIdentifier{}
+	_, err = cs.onlineFetcher.UnsafeTransactions(ctx, emptyNetwork, emptyBlock, emptyTx)
+	validateErrorObject(err, output)
+
+	return output
+}
+
+// Searching for an account backwards from the tip
+func (cs *checkSpec) getAccount(ctx context.Context) (
+	*types.AccountIdentifier,
+	*types.PartialBlockIdentifier,
+	[]*types.Currency,
+	error) {
+	res, err := cs.onlineFetcher.NetworkStatusRetry(ctx, Config.Network, nil)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("%v: unable to get network status", err.Err)
+	}
+
+	var acct *types.AccountIdentifier
+	var blockID *types.PartialBlockIdentifier
+	tip := res.CurrentBlockIdentifier.Index
+	genesis := res.GenesisBlockIdentifier.Index
+	currencies := []*types.Currency{}
+
+	for i := tip; i >= genesis && acct == nil; i-- {
+		blockID = &types.PartialBlockIdentifier{
+			Index: &i,
+		}
+
+		block, err := cs.onlineFetcher.BlockRetry(ctx, Config.Network, blockID)
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("%v: unable to fetch block at index: %v", err.Err, i)
+		}
+
+		// looking for an account in block transactions
+		for _, tx := range block.Transactions {
+			for _, op := range tx.Operations {
+				if op.Account != nil && op.Amount.Currency != nil {
+					acct = op.Account
+					currencies = append(currencies, op.Amount.Currency)
+					break
+				}
+			}
+
+			if acct != nil {
+				break
+			}
+		}
+	}
+
+	return acct, blockID, currencies, nil
+}
+
+func runCheckSpecCmd(_ *cobra.Command, _ []string) error {
+	ctx := context.Background()
+	cs, err := newCheckSpec(ctx)
+	if err != nil {
+		return fmt.Errorf("%v: unable to create checkSpec object with online URL", err)
+	}
+
+	output := []checkSpecOutput{}
+	// validate api endpoints
+	output = append(output, cs.networkStatus(ctx))
+	output = append(output, cs.networkList(ctx))
+	output = append(output, cs.networkOptions(ctx))
+	output = append(output, cs.accountBalance(ctx))
+	output = append(output, cs.accountCoins(ctx))
+	output = append(output, cs.block(ctx))
+	output = append(output, cs.errorObject(ctx))
+	output = append(output, twoModes())
+
+	printInfo("check:spec is complete\n")
+	printCheckSpecOutputHeader()
+	for _, o := range output {
+		printCheckSpecOutputBody(o)
```

Comment:
Do we want to hook up with result.json / reporting service?

Reply:
Great question, I think we should, so our workflow will be able to read the result and publish to test runner. I'll create a new jira for this feature.

Reply:
Jira created: https://jira.coinbase-corp.com/browse/ROSE-408. I'll do it late this week or next week. The logic should be straightforward.